### DISCUSSION TITLE
Do you know that the Power Values are showing Wrong in the OpenDTU with a Battery Operation?

### DISCUSSION THREAD

**Discussion by DrknMonkey** [tbnobody/OpenDTU#1020] [General]
Hello,

i had a Discussion in a German Forum about the Power Values from the DTU (with a Battery Operation instaed of PV Modules).
One Member has tell me that the Power Values in the OpenDTU are showing wrong inclusive in a Original Hoymiles DTU Pro.
So i have checked that and measure these, and it is exact what he has sayed: the Values are wrong. So the showing Kilowatt Hours are also wrong if you use a Limit higher than 200 Watt.

Here a few Pictures what i could measuring:

Until 200 Watt Power Value all is fine and correct. ..
<a href="https://abload.de/image.php?img=img_20230528_1432391mfk6.jpg"><img src="https://abload.de/thumb/img_20230528_1432391mfk6.jpg" /></a>

Already at a 220 Watt Limit you get real 250 Watt Output Power...
<a href="https://abload.de/image.php?img=19150-img202305281436kpdoy.jpg"><img src="https://abload.de/thumb/19150-img202305281436kpdoy.jpg" /></a>

And at 240 Watt you have already reached the maximum Power from 300 Watt...
<a href="https://abload.de/image.php?img=img_20230528_144541ktcc3.jpg"><img src="https://abload.de/thumb/img_20230528_144541ktcc3.jpg" /></a>

A higher Limit up to 300 Watt or 100% brings no Changes more...
<a href="https://abload.de/image.php?img=img_20230528_143049fmfjt.jpg"><img src="https://abload.de/thumb/img_20230528_143049fmfjt.jpg" /></a>

**Comment by Solarteur** [comment]
I can not confirm wrong values. 
The picture shows the current comparison of the values read (!) from the inverters by OpenDTU with the value measured by a Shelly EM.
![Screenshot_20230609](https://github.com/tbnobody/OpenDTU/assets/117977338/6039fdca-df62-4281-b549-31116e16dcac)

**Reply by DrknMonkey** [reply]
Do you use a Battery or PV Operation?

**Reply by Solarteur** [reply]
I am using OpenDTU with a PV system. 
If this error only occurs when operating the inverter with a battery, an appropriately adapted heading would be good.

**Reply by DrknMonkey** [reply]
Ok, maybe this Error is only exist if using the Hoymiles Inverter with a Battery.
I have Change the Heading...

**Reply by DPO99** [reply]
Ich nutze einen HM1500+HM600 an 3 Pylontech us2000c+1x us5000c....und läuft m.E. korrekt.

![image](https://github.com/tbnobody/OpenDTU/assets/125746530/e38c1818-f6e7-44d6-b259-ae6e39afdb1b)

**Comment by DrknMonkey** [comment]
**Interesting... the other Member from another Forum has three HM-300 with a Original DTU Pro and all have wrong Values in the Open DTU. It should can be that "only" the HM-300 have these Error. Now we need someone with an HM-300 + Battery Operation :). My Cerbo GX tells me the same... 300 Watt from the Batterie but the OpenDTU says only 240 Watt Output.

![OpenDTU Cerbo](https://github.com/tbnobody/OpenDTU/assets/127410287/d003bc84-ade6-4dbf-a77d-ea75f267b6fc)

**Reply by DPO99** [reply]
Es besteht noch der unterschied, dass ich ein 48v System habe und du ein 24v. Ich habe schon mehrfach gelesen, dass der HM an 24v nicht rund laufen soll. Aber ob das Damit zusammen hängt...?

M.e. wird der Wert von der Opendtu ja auch nur vom HM ausgelesen.

**Reply by Head** [reply]
Bin hier gelandet, weil ich grade festgestellt hab, warum mein Power-Graph so "nevös" ist: Bis/Unter 200W ist alles schön, aber sobald ich den HM-1500 am 49V geladenen Pylontech auf 250W (oder ähnlich über 200W) stelle schwankt er mit jeder Aktualisierung. 180, 222, 189, 193, 252, 193, 176, ... Am Zähler sieht es ähnlich aus, denke nicht, dass der Wert _falsch_ ist, er schwankt nur einfach wie verrückt.

**Reply by DPO99** [reply]
Ich habe mal bisschen google bemüht. Das Problem scheint bei mehrere aufzutreten...aber nur am Akku. Würde man das Limit mit SolarPanels reduzieren, tritt das Problem wohl nicht auf. Ich habe bisher aber keine Lösung für das Problem finden können.

Ggf. ist das ja ein Spannungsthema....müsste mal jemand mit einem DC-DC Wandler schauen, ob das Problem dann auch auftritt.

**Reply by Head** [reply]
Oder der MPPT Algo vom HM mischt mit rein, aber warum nur über 200W?
Ich tracke jetzt erstmal die DTU Daten in meiner InfluxDB mit, dann kann ich in ein paar Tagen besser auswerten ab wann das passiert. Wer weiß, vielleicht ist es von der Akku-Spannung abhängig oder dergleichen...

**Reply by Head** [reply]
Bis 200 alles recht stabil, drüber deutlich unruhiger:
![image](https://github.com/tbnobody/OpenDTU/assets/51700/c8e66946-3904-4e36-bf62-584bb7520a67)

**Comment by nttmatze** [comment]
Ich habe ein ähnliches Verhalten mit einem HM-600 und auch einer 24V Batterie, ab ca. 400 Watt weichen die Werte von einander ab. Je höher ich die max. leistung erhöhe um so größer sind auch die Differenzen. Komisch ist auch, dass lt. openDTU nicht höher als 485 Watt geht, der eigentlich Output am Shelly ist aber schon bei 650 Watt (was deutlich höher ist wie das was er bringen sollte)
Anbei ein paar Daten zur Beschreibung des Problems:

![grafik](https://github.com/tbnobody/OpenDTU/assets/45540425/2d50c244-0997-414e-9e0a-a666f383710e)
![grafik](https://github.com/tbnobody/OpenDTU/assets/45540425/1c3f334f-ee34-461e-bffd-fe1adc2a4163)
![grafik](https://github.com/tbnobody/OpenDTU/assets/45540425/c198c4e3-2f5d-4d0b-8a73-f6bfe6dd9e75)
![grafik](https://github.com/tbnobody/OpenDTU/assets/45540425/6af6575b-e28e-4d57-8816-73ad13a1f3a8)
![grafik](https://github.com/tbnobody/OpenDTU/assets/45540425/09dd79e0-83d5-4a5f-8297-56b690ebe027)

**Reply by DrknMonkey** [reply]
Yes, thats exact the same Problem just with a Bigger Inverter... and until now it gives no Solution for this Problem.
Thanks for the Measurement and the Pictures ;)

**Reply by cnitschkowski** [reply]
I have the same configuration and the same problem.
I don't measure the output using a Shelly but my regular power meter shows me a higher input than the HM appearently produces.
I also can confirm the discrepancy grows with the actual production...
I have no idea yet how to compensate for this behavior though...

**Comment by Solarteur** [comment]
From my point of view this is less an OpenDTU problem and more a problem with the Hoymiles inverters when operating with a voltage source (battery) for which they are actually not designed. The MPPTs are optimized for operation with PV modules. For example, the internal resistance of the batteries can differ from the internal resistance of the PV modules. From my point of view, this would explain why this "fault" only becomes apparent at higher power levels.

**Reply by cnitschkowski** [reply]
Yes, that must be a problem in the Hoymiles inverter - I just wonder why that is...
I'd guess the voltage on AC phase 1 is correct (otherwise it would cause problems with the public powergrid), so it must be the current on AC phase 1 that is miscalculated for some reason.

**Comment by math322** [comment]
The input battery voltage is for sure a problem, if i run it above 51V everything looks good, if the battery voltage goes below this value the ?MPPT? kicks in and destroys every normal operation. Idk what exactly the problem is, maybe the voltage level but if i run it on a 28V power supply (MW 1kw) it doesn't show this behavior.

**Reply by math322** [reply]
Maybe it's the voltage drift over time.

**Comment by xell66** [comment]
Bei mir das gleiche Problem mit einem HMS-2000 4T und einer 24V Batterie. Mögliche Lösungen:
- MPPT deaktivieren, dafür müsste der Wechselrichter geöffnet werden
- Eine 48V Batterie nutzen
- Einen DC-DC Wandler nutzen und mit der 24V Batterie 48V am Wechselrichter nutzen
- Einen oder mehrere HM300,350 oder 400 nutzen, dort beginnt das MPPT erst über 30 Volt

Falls jemand noch eine schönere Lösung hat, gerne bescheid sagen.

**Reply by DrknMonkey** [reply]
>     * Einen oder mehrere HM300,350 oder 400 nutzen, dort beginnt das MPPT erst über 30 Volt

Bei mir läuft ein einzelner HM-300 und ich habe das Problem ;).

**Reply by kpwg** [reply]
>     * MPPT deaktivieren, dafür müsste der Wechselrichter geöffnet werden
Magst Du das erklären? Gibt es hierzu Erfahrungen?

Ich betreibe einen HM-1200 seit über zwei Jahren mit Batterie. Die übliche Systemspannung bewegt sich dabei zwischen 26 und 28V, nie darüber und selten darunter. Bisher hatte ich den HM-1200 mit nur einem MPPT-Kanal und parallel geschaltenen Eingängen am Laufen, was zwischen 30W und etwa 350W sehr gut funktioniert, darüber wird der Wirkungsgrad rapide schlechter (so bis 400W) und ab etwa 450W ist Schluss, da dann der maximale Strom auf DC-Seite pro Kanal erreicht ist (um die 11A, müsste ich nachsehen). Die MPPT-Regelung habe ich damit spannungsmäßig "unterwandert" und in der Praxis war die Einspeiseleistung auch stets recht konstant.
Nun habe ich den DC-Zweig überarbeitet und mir die beiden anderen Kanäle des HM-1200 parallel dazu geschalten. Plötzlich tritt das oben beschriebene Verhalten auf, in mittleren Bereichen regelt der HM ohne Änderung der Parameter wie wild und erst über 400W wird es wieder ruhig. Ich nutze das Ganze bis 700W, damit ich mit unter 30A noch beherrschbare Ströme habe.

>     * Eine 48V Batterie nutzen

Aus Kostengründen scheidet das aus. Dann kann ich auch einen Multiplus nehmen.

>     * Einen DC-DC Wandler nutzen und mit der 24V Batterie 48V am Wechselrichter nutzen

In der Leistungsklasse muss man schon etwas Geld iin die Hand nehmen. Ich bin auch nicht sicher, ob ein 48V System das MPPT-Problem nicht hat. Jedenfalls kämpft man dort nur mit halb so großen Strömen ;)
